cli/script/command-executor.ts

```diff
@@ -965,8 +966,16 @@ export var releaseReact = (command: cli.IReleaseReactCommand): Promise<void> =>
             throw new Error(`Entry file "${entryFile}" does not exist.`);
         }
     }
+    
+    if (command.appStoreVersion && semver.validRange(command.appStoreVersion) === null) {
+        throw new Error("Please use a semver-compliant target binary version range, for example \"1.0.0\", \"*\" or \"^1.2.3\".");
+    }
```

Comment:
I see this being used in multiple places, may be refactor the above three lines into a separate method `validateAppVersion` which throws error on invalid range.